test/es-module/test-typescript-commonjs.mjs

```diff
@@ -59,15 +59,28 @@ test('require a .ts file with implicit extension fails', async () => {
 });
 
 test('expect failure of an .mts file with CommonJS syntax', async () => {
-  const result = await spawnPromisified(process.execPath, [
-    fixtures.path('typescript/cts/test-cts-but-module-syntax.cts'),
-  ]);
+  const testFilePath = fixtures.path('typescript/cts/test-cts-but-module-syntax.cts');
+  const result = await spawnPromisified(process.execPath, [testFilePath]);
 
-  strictEqual(result.stdout, '');
-  match(result.stderr, /To load an ES module, set "type": "module" in the package\.json or use the \.mjs extension\./);
-  strictEqual(result.code, 1);
+  assert.strictEqual(result.stdout, '');
+
+  const expectedWarning = `Failed to load the ES module: ${testFilePath}. Make sure to set "type": "module" in the nearest package.json file or use the .mjs extension.`;
+
+  try {
+    assert.match(result.stderr, /Failed to load the ES module:.*test-cts-but-module-syntax\.cts/);
+  } catch (e) {
+    if (e?.code === 'ERR_ASSERTION') {
+      e.expected = expectedWarning;
+      e.actual = result.stderr;
+      e.operator = 'includes';
+    }
+    throw e;
+  }
```

Comment:
You don't need the try/catch with `assert.match`
```suggestion
  assert.match(result.stderr, /Failed to load the ES module:.*test-cts-but-module-syntax\.cts/);
```

Reply:
thanks, I updated